lua/bufferline/bbye.lua

```diff
@@ -93,10 +109,10 @@ local bbye = {}
 --- @param action string the command to use to delete the buffer (e.g. `'bdelete'`)
 --- @param force boolean if true, forcefully delete the buffer
 --- @param buffer? integer|string the name of the buffer.
---- @param mods? string the modifiers to the command (e.g. `'verbose'`)
+--- @param mods? string|{[string]: any} the modifiers to the command (e.g. `'verbose'`)
 --- @param focus_id? number the preferred buffer to focus
 function bbye.delete(action, force, buffer, mods, focus_id)
-  local buffer_number = type(buffer) == 'string' and bufnr(buffer) or buffer or get_current_buf()
+  local buffer_number = type(buffer) == 'string' and bufnr(buffer) or tonumber(buffer) or get_current_buf()
```

Comment:
The LSP couldn't figure out that `buffer_number` was always a number unless I put the cast here.